core/src/mender-api.c

```diff
@@ -717,7 +717,11 @@ mender_api_http_artifact_callback(mender_http_client_event_t event, void *data,
     switch (event) {
         case MENDER_HTTP_EVENT_CONNECTED:
             /* Create new artifact context */
-            if (NULL == (mender_artifact_ctx = mender_artifact_create_ctx())) {
+            /* Having the context's internal buffer of size (2 * MENDER_ARTIFACT_STREAM_BLOCK_SIZE +
+               mender_http_recv_buf_length) means we should always have enough space for unprocessed
```

Comment:
Actually, this is not always true it seems. (clever me adding that log message when growing the buffer! :grin: :innocent: )